hudi-common/src/test/java/org/apache/hudi/common/testutils/HoodieTestUtils.java

```diff
@@ -204,6 +204,7 @@ public static HoodieTableMetaClient init(StorageConfiguration<?> storageConf, St
     if (!Objects.equals(keyGen, "org.apache.hudi.keygen.NonpartitionedKeyGenerator")
         && !properties.containsKey("hoodie.datasource.write.partitionpath.field")) {
       builder.setPartitionFields("some_nonexistent_field");
+      builder.setShouldDropPartitionColumns(true);
```

Comment:
I think this should not be added as `hoodie.datasource.write.drop.partition.columns` is originally intended for compatibility with BigQuery only and it is seldomly used; there are other ways of making BigQuery read Hudi tables.  We should likely remove this write/table config.